numpy-stubs/__init__.pyi

```diff
@@ -82,10 +87,39 @@ _DtypeLike = Union[
     Tuple[_DtypeLikeNested, _DtypeLikeNested],
 ]
 
+# Can be used as such `Union[Sequence[T], _ArrayLikeSeq]` as ndarray
+# is not a proper Sequence
+# TODO: Specify the to-be returned array type once ndarray is a Generic
+class _ArrayLikeSeq(Protocol):
```

Comment:
``Union[Sequence[T], _ArrayLikeSeq]`` can be used for annotating sequences, as ``ndarray`` is technically not a proper sequence.
In the future (once ndarray is a generic) this can be changed into ``Union[Sequence[T], _ArrayLikeSeq[T]]``